include/mbgl/util/image.hpp

```diff
@@ -58,6 +71,50 @@ class Image : private util::noncopyable {
     size_t stride() const { return channels * size.width; }
     size_t bytes() const { return stride() * size.height; }
 
+    void fill(uint8_t value) {
+        std::fill(data.get(), data.get() + bytes(), value);
+    }
+
+    // Copy image data within `rect` from `src` to the rectangle of the same size at `pt`
+    // in `dst`. If the specified bounds exceed the bounds of the source or destination,
+    // throw `std::out_of_range`. Must not be used to move data within a single Image.
+    static void copy(const Image& srcImg, Image& dstImg, const Point<uint32_t>& srcPt, const Point<uint32_t>& dstPt, const Size& size) {
```

Comment:
Agreed, I find member functions for this very confusing, since various APIs use different copy directions (e.g. Canvas uses `drawImage` to draw the passed argument into itself, but Node's `Buffer.copy` copies *into* the passed argument...)